components/automatic-points-table/commons/automatic-points-table.lua

```diff
@@ -54,4 +58,76 @@ function AutomaticPointsTable:parseTournaments(args)
 	return tournaments
 end
 
+function AutomaticPointsTable:parseTeams(args)
+	local teams = {}
+	for _, team in Table.iter.pairsByPrefix(args, 'team') do
+		local parsedTeam = Json.parse(team)
+		parsedTeam.aliases = self:parseAliases(parsedTeam)
+		parsedTeam.deductions = self:parseDeductions(parsedTeam)
+		parsedTeam.manualPoints = self:parseManualPoints(parsedTeam)
+		table.insert(teams, parsedTeam)
+	end
+	return teams
+end
+
+--- Parses the team aliases, used in cases where a team is picked up by an org or changed name in some
+--- of the tournaments, in which case aliases are required to correctly query the team's results & points
+function AutomaticPointsTable:parseAliases(team)
+	local aliases = {}
+	for key, value in pairs(team) do
+		if type(key) == 'string' and string.find(key, 'alias%d+') then
+			local aliasIndexString = String.split(key, 'alias')[1]
+			local aliasIndex = tonumber(aliasIndexString)
+			local aliasName = value
+			aliases[aliasIndex] = aliasName
+		end
+	end
+	return aliases
+end
+
+--- Parses the teams' deductions, used in cases where a team has disbanded or made a roster change
+--- that causes them to lose a portion or all of their points that they've accumulated up until that change
+function AutomaticPointsTable:parseDeductions(team)
+	local deductions = {}
+	for key, value in pairs(team) do
+
+		if type(key) == 'string' then
+			if string.find(key, 'deduction%d+note') then
+				local deductionIndex = tonumber(string.match(key, '%d+'))
+				local deductionNote = value

```

Comment:
Use pairsByPrefix iteration

Reply:
It's not guaranteed that if there's **deduction5** there will be **deduction1-deduction4** which means the function from **Table** module wouldn't work

Reply:
Please don't resolve unresolved conversations

Reply:
Conversation addressed here: https://github.com/Liquipedia/Lua-Modules/pull/930#discussion_r782158432